stan/math/prim/scal/fun/is_nan.hpp

```diff
@@ -14,8 +15,24 @@ namespace math {
  * @param x Value to test.
  * @return <code>true</code> if the value is NaN.
  */
-inline bool is_nan(double x) { return std::isnan(x); }
+template <typename T>
+inline bool is_nan(T x) {
+  return std::isnan(x);
+}
 
+/**
+ * Returns true if the input is NaN and false otherwise.
```

Comment:
Can you make this doc more precise? It should indicate that it returns `true` if any of the input is `NaN` (as opposed to all).